mlflow/utils/autologging_utils/__init__.py

```diff
@@ -486,17 +503,38 @@ def autologging_is_disabled(integration_name):
 
 
 @contextlib.contextmanager
-def disable_autologging():
+def disable_autologging(exemptions=None):
     """
     Context manager that temporarily disables autologging globally for all integrations upon
     entry and restores the previous autologging configuration upon exit.
+
+    Args:
+        exemptions: flavors that we do not disable
     """
+    if exemptions is None:
+        exemptions = []
     global _AUTOLOGGING_GLOBALLY_DISABLED
+    global _AUTOLOGGING_GLOBALLY_DISABLED_EXEMPTIONS
     _AUTOLOGGING_GLOBALLY_DISABLED = True
+    _AUTOLOGGING_GLOBALLY_DISABLED_EXEMPTIONS = exemptions
     try:
         yield None
     finally:
         _AUTOLOGGING_GLOBALLY_DISABLED = False
+        _AUTOLOGGING_GLOBALLY_DISABLED_EXEMPTIONS = []
+
+
+@contextlib.contextmanager
+def restrict_langchain_autologging_to_traces_only():
+    if importlib.util.find_spec("langchain") is None:
```

Comment:
Is it sufficient to check `sys.modules` instead? Are there examples that recommend using `find_spec()` instead?

Reply:
There are a few examples throughtout the codebase that use `importlib.util.find_spec`, but I can change it to `sys.modules`